extension/runtime/src/main/java/org/iris_events/producer/EventProducer.java

```diff
@@ -188,15 +257,39 @@ private void doSend(final Object message, final String userId, final boolean pro
 
         switch (scope) {
             case INTERNAL ->
-                publish(message, getRoutingDetailsFromAnnotation(messageAnnotation, scope, userId, propagate));
+                    publish(message, getRoutingDetailsFromAnnotation(messageAnnotation, scope, userId, propagate));
             case USER, SESSION, BROADCAST -> publish(message,
                     getRoutingDetailsForClientScope(messageAnnotation, scope, userId));
             default -> throw new IrisSendException("Message scope " + scope + " not supported!");
         }
     }
 
+    public void sendRpcResponse(final Object message, final String replyTo) {
```

Comment:
Move this from EventProducer, maybe into Consumer code as this will only be called on the RPC consumer side.